From 0d423cbb0342cce48c55f78d3f9fc702e751216d Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Thu, 27 Feb 2020 14:59:22 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399-evb-ind: move the configure for
 right place

Move the common configure are used for rk3399-evb-ind.dtsi.

Change-Id: Ia42ed2bb823d3d30c8433be5bfc07ed48e6b0015
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 .../rk3399-evb-ind-lpddr4-android-avb.dts     | 50 ++++---------------
 .../rk3399-evb-ind-lpddr4-android.dts         | 46 ++---------------
 .../boot/dts/rockchip/rk3399-evb-ind.dtsi     | 45 +++++++++++++++--
 3 files changed, 54 insertions(+), 87 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind-lpddr4-android-avb.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind-lpddr4-android-avb.dts
index f5c3d38f120b..e42b9cdeaa3c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind-lpddr4-android-avb.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind-lpddr4-android-avb.dts
@@ -6,6 +6,7 @@
 /dts-v1/;
 
 #include "rk3399-evb-ind.dtsi"
+#include "rk3399-android.dtsi"
 
 / {
 	model = "Rockchip RK3399 EVB IND LPDDR4 Board edp (Android)";
@@ -19,47 +20,6 @@
 		reg = <0x0 0xff8d0000 0x0 0x20000>;
 	};
 
-	backlight: backlight {
-		status = "okay";
-		compatible = "pwm-backlight";
-		pwms = <&pwm2 0 25000 0>;
-		brightness-levels = <
-			  0   1   2   3   4   5   6   7
-			  8   9  10  11  12  13  14  15
-			 16  17  18  19  20  21  22  23
-			 24  25  26  27  28  29  30  31
-			 32  33  34  35  36  37  38  39
-			 40  41  42  43  44  45  46  47
-			 48  49  50  51  52  53  54  55
-			 56  57  58  59  60  61  62  63
-			 64  65  66  67  68  69  70  71
-			 72  73  74  75  76  77  78  79
-			 80  81  82  83  84  85  86  87
-			 88  89  90  91  92  93  94  95
-			 96  97  98  99 100 101 102 103
-			104 105 106 107 108 109 110 111
-			112 113 114 115 116 117 118 119
-			120 121 122 123 124 125 126 127
-			128 129 130 131 132 133 134 135
-			136 137 138 139 140 141 142 143
-			144 145 146 147 148 149 150 151
-			152 153 154 155 156 157 158 159
-			160 161 162 163 164 165 166 167
-			168 169 170 171 172 173 174 175
-			176 177 178 179 180 181 182 183
-			184 185 186 187 188 189 190 191
-			192 193 194 195 196 197 198 199
-			200 201 202 203 204 205 206 207
-			208 209 210 211 212 213 214 215
-			216 217 218 219 220 221 222 223
-			224 225 226 227 228 229 230 231
-			232 233 234 235 236 237 238 239
-			240 241 242 243 244 245 246 247
-			248 249 250 251 252 253 254 255>;
-		default-brightness-level = <200>;
-		enable-gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
-	};
-
 	vcc_lcd: vcc-lcd {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_lcd";
@@ -112,6 +72,10 @@
 	};
 };
 
+&backlight {
+	enable-gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+};
+
 &dmac_bus {
 	iram = <&iram>;
 	rockchip,force-iram;
@@ -353,6 +317,10 @@
 	status = "okay";
 };
 
+&hdmi_in_vopb {
+	status = "disabled";
+};
+
 &rkisp1_0 {
 	status = "okay";
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind-lpddr4-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind-lpddr4-android.dts
index 69e609ed9243..2b251548d968 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind-lpddr4-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind-lpddr4-android.dts
@@ -6,6 +6,7 @@
 /dts-v1/;
 
 #include "rk3399-evb-ind.dtsi"
+#include "rk3399-android.dtsi"
 
 / {
 	model = "Rockchip RK3399 EVB IND LPDDR4 Board edp (Android)";
@@ -16,46 +17,6 @@
 		reg = <0x0 0xff8d0000 0x0 0x20000>;
 	};
 
-	backlight: backlight {
-		status = "okay";
-		compatible = "pwm-backlight";
-		pwms = <&pwm2 0 25000 0>;
-		brightness-levels = <
-			  0   1   2   3   4   5   6   7
-			  8   9  10  11  12  13  14  15
-			 16  17  18  19  20  21  22  23
-			 24  25  26  27  28  29  30  31
-			 32  33  34  35  36  37  38  39
-			 40  41  42  43  44  45  46  47
-			 48  49  50  51  52  53  54  55
-			 56  57  58  59  60  61  62  63
-			 64  65  66  67  68  69  70  71
-			 72  73  74  75  76  77  78  79
-			 80  81  82  83  84  85  86  87
-			 88  89  90  91  92  93  94  95
-			 96  97  98  99 100 101 102 103
-			104 105 106 107 108 109 110 111
-			112 113 114 115 116 117 118 119
-			120 121 122 123 124 125 126 127
-			128 129 130 131 132 133 134 135
-			136 137 138 139 140 141 142 143
-			144 145 146 147 148 149 150 151
-			152 153 154 155 156 157 158 159
-			160 161 162 163 164 165 166 167
-			168 169 170 171 172 173 174 175
-			176 177 178 179 180 181 182 183
-			184 185 186 187 188 189 190 191
-			192 193 194 195 196 197 198 199
-			200 201 202 203 204 205 206 207
-			208 209 210 211 212 213 214 215
-			216 217 218 219 220 221 222 223
-			224 225 226 227 228 229 230 231
-			232 233 234 235 236 237 238 239
-			240 241 242 243 244 245 246 247
-			248 249 250 251 252 253 254 255>;
-		default-brightness-level = <200>;
-	};
-
 	vcc_lcd: vcc-lcd {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_lcd";
@@ -109,7 +70,6 @@
 };
 
 &backlight {
-	status = "okay";
 	enable-gpios = <&gpio1 18 GPIO_ACTIVE_HIGH>;
 };
 
@@ -163,6 +123,10 @@
 	status = "okay";
 };
 
+&hdmi_in_vopb {
+	status = "disabled";
+};
+
 /*
  * if enable dp_sound, should disable spdif_sound and spdif_out
  */
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
index 9d37f3c67b1b..578c3aaf9fe7 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-ind.dtsi
@@ -5,7 +5,6 @@
 
 #include "dt-bindings/pwm/pwm.h"
 #include "rk3399.dtsi"
-#include "rk3399-android.dtsi"
 #include "rk3399-opp.dtsi"
 #include <dt-bindings/sensor-dev.h>
 #include "rk3399-vop-clk-set.dtsi"
@@ -34,6 +33,46 @@
 		};
 	};
 
+	backlight: backlight {
+		status = "okay";
+		compatible = "pwm-backlight";
+		pwms = <&pwm2 0 25000 0>;
+		brightness-levels = <
+			  0   1   2   3   4   5   6   7
+			  8   9  10  11  12  13  14  15
+			 16  17  18  19  20  21  22  23
+			 24  25  26  27  28  29  30  31
+			 32  33  34  35  36  37  38  39
+			 40  41  42  43  44  45  46  47
+			 48  49  50  51  52  53  54  55
+			 56  57  58  59  60  61  62  63
+			 64  65  66  67  68  69  70  71
+			 72  73  74  75  76  77  78  79
+			 80  81  82  83  84  85  86  87
+			 88  89  90  91  92  93  94  95
+			 96  97  98  99 100 101 102 103
+			104 105 106 107 108 109 110 111
+			112 113 114 115 116 117 118 119
+			120 121 122 123 124 125 126 127
+			128 129 130 131 132 133 134 135
+			136 137 138 139 140 141 142 143
+			144 145 146 147 148 149 150 151
+			152 153 154 155 156 157 158 159
+			160 161 162 163 164 165 166 167
+			168 169 170 171 172 173 174 175
+			176 177 178 179 180 181 182 183
+			184 185 186 187 188 189 190 191
+			192 193 194 195 196 197 198 199
+			200 201 202 203 204 205 206 207
+			208 209 210 211 212 213 214 215
+			216 217 218 219 220 221 222 223
+			224 225 226 227 228 229 230 231
+			232 233 234 235 236 237 238 239
+			240 241 242 243 244 245 246 247
+			248 249 250 251 252 253 254 255>;
+		default-brightness-level = <200>;
+	};
+
 	clkin_gmac: external-gmac-clock {
 		compatible = "fixed-clock";
 		clock-frequency = <125000000>;
@@ -397,10 +436,6 @@
 	status = "okay";
 };
 
-&hdmi_in_vopb {
-	status = "disabled";
-};
-
 &i2c0 {
 	status = "okay";
 	i2c-scl-rising-time-ns = <168>;
-- 
2.35.3

